src/inputpanel/QskInputContext.cpp

```diff
@@ -494,18 +500,62 @@ void QskInputContext::commitPrediction( bool )
      */
 }
 
+class QskInputContextFactory::PrivateData
+{
+  public:
+    QThread* thread = nullptr;
+    std::shared_ptr< QskTextPredictor > predictor;
```

Comment:
I made this a shared pointer to support the case where an input panel outlives the predictor (e.g. when an input panel is still open and somebody switches the language and thus deletes the current predictor).
Also, I guess it is debatable whether we should have the methods related to the predictor in the input context factor; the naming sounds a bit misleading...